lib/diff.eex

```diff
@@ -0,0 +1,141 @@
+<style>
+  .viz-tool table {
+    border-collapse: collapse;
+    text-align: left;
+  }
+  .viz-tool td, .viz-tool th {
+    border-bottom: 1px solid #ddd;
+    margin: 0;
+    padding: 0;
+  }
+  .viz-tool .viz-comment {
+    border: 1px solid #ddd;
+    height: 50px;
+    margin-top: 1em;
+    padding: 0.5em;
+  }
+  .viz-tool table.viz-ladder {
+    width: 100%;
+  }
+  .viz-tool table.viz-all-vehicles td, .viz-tool table.viz-all-vehicles th{
+    padding: 0 1em;
+  }
+  .viz-tool .viz-position {
+    width: 14%;
+  }
+  .viz-tool.viz-hidden {
+    display: none;
+  }
+</style>
+
+<button id="viz-tool-previous-button">Previous</button>
+<button id="viz-tool-next-button">Next</button>
+
+<%= for {{{comment_base, by_stop_base}, {comment_diff, by_stop_diff}}, i} <- Enum.with_index(@vehicle_archives) do %>
+  <div id="viz-<%= i %>" class="viz-tool viz-hidden">
+    <div class="viz-comment"><%= comment_base %> / <%= comment_diff %></div>
+
+    <h2>Routes</h2>
+
+    <%= for {route_name, stops} <- @routes do %>
+      <h3><%= route_name %></h3>
+      <table class="viz-ladder">
+        <tr>
+          <th class="viz-position">Transit
+          <th class="viz-position">Incoming</th>
+          <th class="viz-position">At</th>
+          <th>Station</th>
+          <th class="viz-position">At</th>
+          <th class="viz-position">Incoming</th>
+          <th class="viz-position">Transit</th>
+        </tr>
+
+        <%= for {stop_name, stop_id_0, stop_id_1} <- stops do %>
+          <% vs0_base = by_stop_base[stop_id_0] || [] %>
+          <% vs1_base = by_stop_base[stop_id_1] || [] %>
+
+          <% vs0_diff = by_stop_diff[stop_id_0] || [] %>
+          <% vs1_diff = by_stop_diff[stop_id_1] || [] %>
+
+          <tr>
+            <td>
+              <%= Phoenix.HTML.raw trainify_diff(vs0_base, vs0_diff, :IN_TRANSIT_TO, "🚂", "👻")%>
+            </td>
+            <td>
+              <%= Phoenix.HTML.raw trainify_diff(vs0_base, vs0_diff, :INCOMING_AT, "🚂", "👻")%>
+            </td>
+            <td>
+              <%= Phoenix.HTML.raw trainify_diff(vs0_base, vs0_diff, :STOPPED_AT, "🚂", "👻")%>
+            </td>
+            <td>
+              <%= stop_name %>
+            </td>
+            <td>
+              <%= Phoenix.HTML.raw trainify_diff(vs1_base, vs1_diff, :IN_TRANSIT_TO, "🚂", "👻")%>
```

Comment:
I believe in the other template the `:IN_TRANSIT_TO` and `:STOPPED_AT` are reversed, so that `:STOPPED_AT` is always adjacent to the stop name.